src/main/groovy/org/assertj/generator/gradle/internal/tasks/AssertionsGeneratorReport.groovy

```diff
@@ -44,127 +45,135 @@ class AssertionsGeneratorReport {
         userTemplates = new ArrayList<>()
     }
 
-    void setDirectoryPathWhereAssertionFilesAreGenerated(File directory) {
-        this.directoryPathWhereAssertionFilesAreGenerated = directory
-    }
-
     void addGeneratedAssertionFile(File generatedCustomAssertionFile) throws IOException {
         generatedCustomAssertionFileNames.add(generatedCustomAssertionFile.getCanonicalPath())
     }
 
     String getReportContent() {
-        StringBuilder reportBuilder = new StringBuilder(System.lineSeparator())
-        reportBuilder.append(System.lineSeparator())
-        reportBuilder.append("====================================\n")
-        reportBuilder.append("AssertJ assertions generation report\n")
-        reportBuilder.append("====================================\n")
-        buildGeneratorParametersReport(reportBuilder)
-        reportBuilder.append(System.lineSeparator())
-        reportBuilder.append(SECTION_START).append("Generator results").append(SECTION_END)
-        if (generationError()) {
-            buildGeneratorReportError(reportBuilder)
-        } else if (nothingGenerated()) {
-            buildGeneratorReportWhenNothingWasGenerated(reportBuilder)
-        } else {
-            buildGeneratorReportSuccess(reportBuilder)
-        }
-        return reportBuilder.toString()
-    }
-
-    private void buildGeneratorReportSuccess(StringBuilder reportBuilder) {
-        reportBuilder.append(System.lineSeparator())
-        reportBuilder.append("Directory where custom assertions files have been generated:\n")
-        reportBuilder.append(INDENT).append(directoryPathWhereAssertionFilesAreGenerated).append(System.lineSeparator())
-        reportBuilder.append(System.lineSeparator())
-        reportBuilder.append("Custom assertions files generated:\n")
-        for (String fileName : generatedCustomAssertionFileNames) {
-            reportBuilder.append(INDENT).append(fileName).append(System.lineSeparator())
-        }
-        if (!inputClassesNotFound.isEmpty()) {
-            reportBuilder.append(System.lineSeparator())
-            reportBuilder.append("No custom assertions files generated for the following input classes as they were not found:\n")
-            for (String inputClassNotFound : inputClassesNotFound) {
-                reportBuilder.append(INDENT).append(inputClassNotFound).append(System.lineSeparator())
+        new StringBuilder().tap {
```

Comment:
The `tap` method is identical to Kotlin's `apply`. Ideally, I would rewrite the methods that add more content as `Extension` functions, but we can't because Groovy has a limitation where extensions can only be `static` and defined on separate objects. More or less a needless limitation from a language standpoint, but I'm sure there's a reason technically. Kotlin doesn't have this limitation 👍🏻